File: practiceSolvative/app/src/main/java/com/example/practicesolvative/MainActivity.java
package com.example.practicesolvative;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.widget.LinearLayout;

import com.example.practicesolvative.adapter.Adapter;
import com.example.practicesolvative.model.ContactModel;
import com.example.practicesolvative.model.model;
import com.example.practicesolvative.viewmodel.ContactListViewModel;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private List<ContactModel> contactModelList;
    private Adapter adapter;
    private ContactListViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        RecyclerView recyclerView = findViewById(R.id.recyclerview);
        LinearLayoutManager  layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        adapter = new Adapter(this, contactModelList);
        viewModel = ViewModelProviders.of(this).get(ContactListViewModel.class);
//        viewModel.getContactList().observe(this, new Observer<List<ContactModel>>() {
//            @Override
//            public void onChanged(List<ContactModel> contactModels) {
//                if(contactModels != null) {
//                    contactModelList = contactModels;
//                    adapter.setContactModelList(contactModels);
//                }
//            }
//        });
        viewModel.getContactList().observe(this, new Observer<model>() {
            @Override
            public void onChanged(model model) {
                if(model != null){
                    contactModelList = model.getContacts();
                    adapter.setContactModelList(model.getContacts());
                }
            }
        });
        viewModel.makeCall();
    }
}

File: KotlinPractice/app/src/main/java/com/example/kotlinpractice/MainActivity.kt
package com.example.kotlinpractice

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.kotlinpractice.adapter.RecyclerViewAdapter
import com.example.kotlinpractice.model.RepositoryData
import com.example.kotlinpractice.viewmodel.MainActivityViewModel
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    private lateinit var recyclerViewAdapter:RecyclerViewAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViewModel()
        initMainViewModel()

    }
    private fun initViewModel(){
        val recyclerview = findViewById<RecyclerView>(R.id.recycler)
        recyclerview.layoutManager = LinearLayoutManager(this)
        val decoration = DividerItemDecoration(applicationContext,DividerItemDecoration.VERTICAL)
        recyclerview.addItemDecoration(decoration)
        recyclerViewAdapter = RecyclerViewAdapter()
        val adapter = recyclerViewAdapter
        recyclerview.adapter = adapter
    }

    private fun initMainViewModel(){
        val viewModel = ViewModelProvider(this).get(MainActivityViewModel::class.java)
        viewModel.getAllRepositoryList().observe(this, Observer<List<RepositoryData>> {
            recyclerViewAdapter.setListData(it)
            recyclerViewAdapter.notifyDataSetChanged()
        })
        viewModel.makeAPICall()
    }
}